tools/pyi/gen_pyi.py

```diff
@@ -237,14 +200,15 @@ def sig_for_ops(opname: str) -> list[str]:
     assert opname.endswith("__") and opname.startswith("__"), f"Unexpected op {opname}"
 
     name = opname[2:-2]
-    if name in binary_ops:
+    if name in symmetric_comparison_ops:
+        # unsafe override https://github.com/python/mypy/issues/5704
+        return [
+            f"def {opname}(self, other: Any) -> Tensor: ...  # type: ignore[override] # noqa: PYI032"
+        ]
+    if name in inplace_binary_ops:
+        return [f"def {opname}(self, other: Any) -> Self: ..."]
```

Comment:
The core change in this PR is here. Split `inplace_binary_ops` from `binary_ops` and handle them separately.

```python
def __add__(self, other: Any) -> Tensor: ...
def __iadd__(self, other: Any) -> Self: ...
```